prov/util/src/util_buf.c

```diff
@@ -120,11 +120,10 @@ int util_buf_grow(struct util_buf_pool *pool)
 		if (pool->attr.init) {
 #if ENABLE_DEBUG
 			if (!pool->attr.indexing.ordered) {
-				buf_ftr->entry.slist.next = (void *) OFI_MAGIC_64;
-
+				buf_ftr->entry.dlist.next = (void *) OFI_MAGIC_64;
 				pool->attr.init(pool->attr.ctx, buf);
 
-				assert(buf_ftr->entry.slist.next == (void *) OFI_MAGIC_64);
+				assert(buf_ftr->entry.dlist.next == (void *) OFI_MAGIC_64);
```

Comment:
please, try "./configure --enable-debug && make -j 32 && make install"
I think, this doesn't work due to missed `if (!ordered) { if (!garbage_collect) { ... } else { .. } } else { ... }` construction here as we have in other places